chain/postgresdb/pgdb/pgdb_test.go

```diff
@@ -0,0 +1,270 @@
+package pgdb_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/stretchr/testify/require"
+
+	"github.com/drand/drand/chain"
+	chainerrors "github.com/drand/drand/chain/errors"
+	"github.com/drand/drand/chain/postgresdb/pgdb"
+	"github.com/drand/drand/log"
+	"github.com/drand/drand/test"
+)
+
+var c *test.Container
+
+func TestMain(m *testing.M) {
+	var err error
+	c, err = test.StartDB()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	defer test.StopDB(c)
+
+	m.Run()
+}
+
+func Test_OrderStorePG(t *testing.T) {
+	ctx := context.Background()
+	l, db, teardown := test.NewUnit(t, c, t.Name())
+	defer t.Cleanup(teardown)
+
+	beaconName := "beacon"
+	store, err := pgdb.NewStore(context.Background(), l, db, beaconName)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, store.Close(ctx))
+	}()
+
+	b1 := &chain.Beacon{
+		PreviousSig: []byte("a magnificent signature"),
+		Round:       145,
+		Signature:   []byte("one signature to"),
+	}
+
+	b2 := &chain.Beacon{
+		PreviousSig: []byte("is not worth an invalid one"),
+		Round:       146,
+		Signature:   []byte("govern them all"),
+	}
+
+	// we store b2 and check if it is last
+	require.NoError(t, store.Put(ctx, b2))
+	eb2, err := store.Last(ctx)
+	require.NoError(t, err)
+	require.Equal(t, b2, eb2)
+	eb2, err = store.Last(ctx)
+	require.NoError(t, err)
+	require.Equal(t, b2, eb2)
+
+	// then we store b1
+	require.NoError(t, store.Put(ctx, b1))
+
+	// and request last again
+	eb2, err = store.Last(ctx)
```

Comment:
this makes sense, but now I wonder what the bolt store does for this scenario :p

Reply:
I too am surprised about this working as it does, but since the test is lifted from the BoltDB implementation, I just had to make sure that the implementation passes without changing the logic.

I forgot to raise a comment about this, nice catch.

Reply:
Closing because it's working as intended.